plugins/splitbutton/plugin.js

```diff
@@ -94,12 +115,12 @@
 
 			editor.addFeature( properties.buttons[ item.id ] );
 
-			// First button as default. It might be overwritten by actual default button.
-			if ( !defaultButton && !item[ 'default' ] ) {
+			// If no default button was defined then first button will act as default.
```

Comment:
This logic could be now simplified. Here, in the loop, you could just leave `if ( definition[ 'default' ] && definition[ 'default' ] === item.name ) {` check over here.

Then the second `if` should go outside the loop, and that would check for conditions:
* whether `defaultButton` was not picked, with the logic above ☝️
* whether `definition[ 'default' ]` was not defined at all

Then assign the first item as a default button (if there was at least one item).